packages/svelte/src/internal/client/reactivity/cache.js

```diff
@@ -0,0 +1,84 @@
+/** @import { CacheEntry } from '#shared' */
+import { async_mode_flag } from '../../flags/index.js';
+import { BaseCacheObserver } from '../../shared/cache-observer.js';
+import { tick } from '../runtime.js';
+import { get_effect_validation_error_code, render_effect } from './effects.js';
+import * as e from '../errors.js';
+
+/** @typedef {{ count: number, item: any }} Entry */
+/** @type {Map<string, CacheEntry>} */
+const client_cache = new Map();
+
+/**
+ * @template {(...args: any[]) => any} TFn
+ * @param {string} key
+ * @param {TFn} fn
+ * @returns {ReturnType<TFn>}
+ */
+export function cache(key, fn) {
+	if (!async_mode_flag) {
+		e.experimental_async_required('cache');
+	}
+
+	const cached = client_cache.has(key);
+	const entry = client_cache.get(key);
```

Comment:
given that `entry` is an object, if it exists, I don't think we need to use `has`?

```suggestion
	const entry = client_cache.get(key);
```